### ISSUE TITLE
Feedback submit page: Submit only answers that are changed to the back-end

### ISSUE THREAD

**Issue by wkurniawan07** [TEAMMATES/teammates#10605]
In feedback submit page, responses for all questions are submitted (to the back-end for processing) regardless of whether any are changed or not. We can improve the performance of this page by submitting only changed answers.

The difficulty of this task is mainly on determining whether a response has "changed" or not.

**Comment by xpdavid** [comment]
Just a note here. I have brought this idea to @damithc but he suggested it might cause confusions among users: If students open two submission windows (e.g. one in their phone and one in their computer), they would assume the last submission will be their final submission but it might not always be the case when the feature is up.

This is why I did optimization in the DB layer instead to write only when changed.

**Comment by wkurniawan07** [comment]
@xpdavid I think you'll find it interesting that #7341 is an issue

**Comment by antoine-dcl** [comment]
Hello, I would like to tackle this issue (my first one). 
Is it OK ? Thanks.

**Comment by bezalelBU** [comment]
Hello, I would like to work on this issue.

**Comment by rrtheonlyone** [comment]
Hi @bezalelBU, how do you plan on going about with this issue? You might want to start off with our first timer issues and move on to this to get some familiarity with our application.

**Comment by bezalelBU** [comment]
@rrtheonlyone, thank you for responding. I don't have any experience outside of my own projects, so I think you are probably right, and I will look for an issue with that label.

My idea had been to try to modify the script in "...\teammates\src\web\app\pages-session\session-submission-page\session-submission-page.component.ts". Since it checks the length of the responses to make sure they're not empty, I thought it might be a good place to see if the user had modified anything, and only update the db with responses that were changed. Would that have worked?

**Comment by rrtheonlyone** [comment]
@bezalelBU yes that is a good starting point. However, there are multiple other considerations here e.g. we may have to change the backend logic. We have to discuss further on the pros/cons of having this feature too (based off the above discussion)

**Comment by wkurniawan07** [comment]
Close as the gain is no longer significant, now that we allow per-question response submission.
